backend/src/Listener/PokemonListener.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Listener;
+
+use App\Entity\Pokemon;
+use App\Entity\Ability;
+use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+
+class PokemonListener
+{
+    public function postPersist(LifecycleEventArgs $args)
+    {
+        $entity = $args->getObject();
+
+        if (!$entity instanceof Pokemon) {
+            return;
+        }
+
+        if (count($entity->getAbilities()) == 0) {
+            $entityManager = $args->getObjectManager();
+            $abilities = $entityManager->getRepository(Ability::class)->findAll();
```

Comment:
Unfortunately, Doctrine doesn't allow us to select directly a random value.
If we are regarding about performance, we could choose one of the solution in the following link [https://stackoverflow.com/questions/10762538/how-to-select-randomly-with-doctrine](https://stackoverflow.com/questions/10762538/how-to-select-randomly-with-doctrine)